dev/buildtool/flow_generate.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Usage:  spinnaker/dev/buildtool/flow_generate.sh <defaults-file>
+#   e.g.  spinnaker/dev/buildtool/flow_generate.sh ga-defaults.yml
+#
+# Note: This can be run from any directory, however root_dir path
+#       in the <defaults-file> will be relative to the CWD so it
+#       likely makes sense to run from the project parent directory.
+
+set -e
+
+# Import support functions
+source $(dirname $0)/support_functions.sh
+
+
+########################################################
+# Fetches the main sources (wait for completion)
+#
+#   Some specialized source repositories might not be
+#   included here, particularly if they are only used
+#   by individual command modules.
+########################################################
+function run_fetch_flow() {
+  start_command_unless NO_FETCH "fetch_source"
+
+  # Synchronize here so we have source before we continue.
+  wait_for_commands_or_die "Fetch"
+}
+
+
+########################################################
+# Generate all the release artifacts (wait for completion)
+#
+#    This flow generates all the artifacts so they can
+#    be verified and validated. It does not publish any
+#    of them beyond the needs for validation processes.
+########################################################
+function run_generate_flow() {
+  # Not sure if this is needed or if it is covered by build_halyard
+  # start_command_unless NO_HALYARD "build_halyard_containers"
+
+  start_command_unless NO_HALYARD "build_halyard"
+  start_command_unless NO_CONTAINERS "build_bom_containers"
```

Comment:
Which builds the debians?

Reply:
build_halyard seems to build everything (I say seems because it wasnt clear how it is supposed to work because of the release/ scripts and I havent run halyard from a container yet, only on a VM).